### PR TITLE
Standardize Rewards engine unit tests

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/38125

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_rewards/core/wallet_provider/uphold/connect_uphold_wallet.cc [hunk 1]
```diff
@@ -127,15 +123,9 @@ void ConnectUpholdWallet::OnGetUser(ConnectExternalWalletCallback callback,
...
-      base::BindOnce(
-          // NOLINTNEXTLINE
-          static_cast<void (ConnectUpholdWallet::*)(
-              ConnectExternalWalletCallback, const std::string&,
-              const std::string&, mojom::Result, internal::uphold::Capabilities)
-                          const>(&ConnectUpholdWallet::OnGetCapabilities),
-          base::Unretained(this), std::move(callback), access_token,
-          user.country_id));
+      access_token, base::BindOnce(&ConnectUpholdWallet::OnGetCapabilities,
+                                   base::Unretained(this), std::move(callback),
```
[the remark this answers is not in the document]
Comment (zenparsing) [reply components/brave_rewards/core/wallet_provider/uphold/connect_uphold_wallet.cc:127]: Reformatting of existing code.

File: components/brave_rewards/core/wallet_provider/uphold/connect_uphold_wallet.cc [hunk 2]
```diff
@@ -82,18 +82,14 @@
 
   server_.get_me().Request(
       access_token,
-      // NOLINTNEXTLINE
-      base::BindOnce(
-          static_cast<void (ConnectUpholdWallet::*)(
-              ConnectExternalWalletCallback, const std::string&, mojom::Result,
-              const User&) const>(&ConnectUpholdWallet::OnGetUser),
-          base::Unretained(this), std::move(callback), access_token));
+      base::BindOnce(&ConnectUpholdWallet::OnGetUser, base::Unretained(this),
```
[the remark this answers is not in the document]
Comment (zenparsing) [reply components/brave_rewards/core/wallet_provider/uphold/connect_uphold_wallet.cc:85]: Reformatting of existing code.

File: components/brave_rewards/core/test/rewards_engine_test.h [hunk 3]
```diff
@@ -35,24 +38,39 @@ class RewardsEngineTest : public testing::Test {
...
 
   // Returns the |RewardsEngine| instance for this test.
   RewardsEngine& engine() { return engine_; }
 
-  // Adds a mock network response for the specified URL and HTTP method.
-  void AddNetworkResultForTesting(const std::string& url,
-                                  mojom::UrlMethod method,
-                                  mojom::UrlResponsePtr response);
+  // Executes the supplied lambda with an appropriate callback, waits until the
+  // callback has been executed, and the returns the value to the caller.
```
Comment (emerick) [inline components/brave_rewards/core/test/rewards_engine_test.h:47]: Nit: "then returns"

File: components/brave_rewards/core/test/rewards_engine_test.h [hunk 4]
```diff
@@ -35,24 +38,39 @@ class RewardsEngineTest : public testing::Test {
...
+  void WaitFor(F fn) {
+    base::RunLoop run_loop;
+    fn(base::BindLambdaForTesting([&run_loop]() { run_loop.Quit(); }));
+    run_loop.Run();
+  }
 
-  // Sets a callback that is executed when a message is logged to the client.
-  void SetLogCallbackForTesting(TestRewardsEngineClient::LogCallback callback);
+  // Executes the supplied lambda with an appropriate callback, waits until the
+  // callback has been executed, and the returns the value to the caller.
```
Comment (emerick) [inline components/brave_rewards/core/test/rewards_engine_test.h:56]: Same nit :-)

**Review by emerick — APPROVED** [review]
LGTM, couple of minor typo nits

**Pull request merged**